src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceCheckedListBox.cs

```diff
@@ -136,10 +136,11 @@ protected override void OnDrawItem(DrawItemEventArgs e)
                         graphics.DrawString(currentItem.Name, e.Font, foreBrush, nameBounds, drawFormat);
 
                         var packageSource = new PackageSource(currentItem.Source, currentItem.Name);
-                        var isSourceHttp = packageSource.IsHttp && !packageSource.IsHttps;
+                        packageSource.AllowInsecureConnections = currentItem.AllowInsecureConnections;
```

Comment:
Changes look good, but can you help me understand why we need this line?

Reply:
Thanks for reviewing! The `currentItem` is a `PackageSourceContextInfo` with the right `AllowInsecureConnections` from settings. However, it doesn't have `IsHttp `or `IsHttps`.  The `packageSource` is initiated with `Source` and `Name `, so the `AllowInsecureConnections` is of default value which is always false. So it needs to get `AllowInsecureConnections` from `currentItem`.

Reply:
Thanks! You're right, that looks much better :)